src/lib.rs

```diff
@@ -1021,6 +1046,338 @@ safety_comment! {
     unsafe_impl_known_layout!(T: ?Sized + KnownLayout => #[repr(T::MaybeUninit)] MaybeUninit<T>);
 }
 
+mod split_at {
+    use super::*;
+
+    /// Types that can be split in two.
+    ///
+    /// # Implementation
+    ///
+    /// **Do not implement this trait yourself!** Instead, use
+    /// [`#[derive(SplitAt)]`][derive]; e.g.:
+    ///
+    /// ```
+    /// # use zerocopy_derive::{SplitAt, KnownLayout};
+    /// #[derive(SplitAt, KnownLayout)]
+    /// #[repr(C)]
+    /// struct MyStruct<T: ?Sized> {
+    /// # /*
+    ///     ...,
+    /// # */
+    ///     field: T,
+    /// }
+    /// ```
+    ///
+    /// This derive performs a sophisticated, compile-time safety analysis to
+    /// determine whether a type is `SplitAt`.
+    ///
+    /// # Safety
+    ///
+    /// This trait does not convey any safety guarantees to code outside this crate.
+    ///
+    /// You must not rely on the `#[doc(hidden)]` internals of `SplitAt`. Future
+    /// releases of zerocopy may make backwards-breaking changes to these items,
+    /// including changes that only affect soundness, which may cause code which
+    /// uses those items to silently become unsound.
+    ///
+    #[cfg_attr(feature = "derive", doc = "[derive]: zerocopy_derive::SplitAt")]
+    #[cfg_attr(
+        not(feature = "derive"),
+        doc = concat!("[derive]: https://docs.rs/zerocopy/", env!("CARGO_PKG_VERSION"), "/zerocopy/derive.SplitAt.html"),
+    )]
+    #[cfg_attr(
+        zerocopy_diagnostic_on_unimplemented_1_78_0,
+        diagnostic::on_unimplemented(note = "Consider adding `#[derive(SplitAt)]` to `{Self}`")
+    )]
+    // SAFETY: The trailing slice is well-aligned for its element type.
```

Comment:
```suggestion
    // # Safety
    //
    // The trailing slice must be well-aligned for its element type.
```

To make it clear that this is an invariant, not a safety proof.

Reply:
https://github.com/google/zerocopy/compare/101bbb6a420aac10d8bf4312305e10f919a3a294..ddb594e5907f90885faebb3e27c6eeb2a1797a0a#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R1092-R1094